src/egon/data/datasets/power_plants/pv_rooftop_buildings.py

```diff
@@ -0,0 +1,2433 @@
+        )
+
+        overlay_gdf.loc[nuts_gdf.index] = overlay_gdf.loc[
+            nuts_gdf.index
+        ].assign(capacity=capacity.multiply(conversion).to_numpy())
+
+    return overlay_gdf[["bus_id", "capacity"]].groupby("bus_id").sum()
+
+
+def determine_end_of_life_gens(
+    mastr_gdf: gpd.GeoDataFrame,
+    scenario_timestamp: pd.Timestamp,
+    pv_rooftop_lifetime: pd.Timedelta,
+) -> gpd.GeoDataFrame:
+    """
+    Determine if an old PV system has reached its end of life.
+    Parameters
+    -----------
+    mastr_gdf : geopandas.GeoDataFrame
+        GeoDataFrame containing geocoded MaStR data.
+    scenario_timestamp : pandas.Timestamp
+        Timestamp at which the scenario takes place.
+    pv_rooftop_lifetime : pandas.Timedelta
+        Average expected lifetime of PV rooftop systems.
+    Returns
+    -------
+    geopandas.GeoDataFrame
+        GeoDataFrame containing geocoded MaStR data and info if the system
+        has reached its end of life.
+    """
+    mastr_gdf = mastr_gdf.assign(
+        age=scenario_timestamp - mastr_gdf.start_up_date
+    )
+
+    logger.debug("Determined if pv rooftop systems reached their end of life.")
+
+    return mastr_gdf.assign(end_of_life=pv_rooftop_lifetime < mastr_gdf.age)
+
+
+def calculate_max_pv_cap_per_building(
+    buildings_gdf: gpd.GeoDataFrame,
+    mastr_gdf: gpd.GeoDataFrame,
+    pv_cap_per_sq_m: float | int,
+    roof_factor: float | int,
+) -> gpd.GeoDataFrame:
+    """
+    Calculate the estimated maximum possible PV capacity per building.
+    Parameters
+    -----------
+    buildings_gdf : geopandas.GeoDataFrame
+        GeoDataFrame containing OSM buildings data.
+    mastr_gdf : geopandas.GeoDataFrame
+        GeoDataFrame containing geocoded MaStR data.
+    pv_cap_per_sq_m : float, int
+        Average expected, installable PV capacity per square meter.
+    roof_factor : float, int
+        Average for PV usable roof area share.
+    Returns
+    -------
+    geopandas.GeoDataFrame
+        GeoDataFrame containing OSM buildings data with estimated maximum PV
+        capacity.
+    """
+    gdf = (
+        buildings_gdf.reset_index()
+        .merge(
+            mastr_gdf[
+                [
+                    "capacity",
+                    "end_of_life",
+                    "building_id",
+                    "EinheitlicheAusrichtungUndNeigungswinkel",
+                    "Hauptausrichtung",
+                    "HauptausrichtungNeigungswinkel",
+                ]
+            ],
+            how="left",
+            left_on="id",
+            right_on="building_id",
+        )
+        .set_index("id")
+        .drop(columns="building_id")
+    )
+
+    gdf = gdf.assign(
+        max_cap=gdf.building_area.multiply(roof_factor * pv_cap_per_sq_m),
+        end_of_life=gdf.end_of_life.fillna(True).astype(bool),
+        bus_id=gdf.bus_id.astype(int),
+    )
+
+    return gdf
+
+
+def calculate_building_load_factor(
```

Comment:
If you obtain the specific capacity factor here, I don't fully understand why you need the hard coded `PV_CAP_PER_SQ_M` above?

Reply:
With `PV_CAP_PER_SQ_M` I calculate the theoretical maximum pv cap per roof from which I derive the `load_factor` (actual / max capacity) calculated within this function.